### ISSUE TITLE
Adding many tools at one time causes MaxListenersExceededWarning

### ISSUE THREAD

**Issue by caphosra** [capra314cabra/Yatsugi#33]
Error logs:
```
(node:3088) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 database-add-tool-reply listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
(node:3088) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 database-lent-tool-reply listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
```

**Comment by caphosra** [comment]
I think this problem will be closed when the feature of #35 is achieved.

**Comment by caphosra** [comment]
Solved it.
